codegenerator/cli/templates/static/codegen/src/EventFetching.res

```diff
@@ -122,26 +122,63 @@ let convertLogs = (
 
   logs->Belt.Array.map(async (log): Types.eventBatchQueueItem => {
     let block = (await blockLoader->LazyLoader.get(log.blockNumber))->blockFieldsFromBlock
-    let (event, eventMod) = switch Converters.parseEvent(
-      ~log=log->ethersLogToLog,
-      ~block,
-      ~eventModLookup,
-      ~contractInterfaceManager,
-      ~chain,
-      ~transaction=log->transactionFieldsFromLog(~logger),
+    let transaction = log->transactionFieldsFromLog(~logger)
+    let log = log->ethersLogToLog
+    let chainId = chain->ChainMap.Chain.toChainId
+
+    let topic0 = log.topics->Js.Array2.unsafe_get(0)
+    let eventMod = switch eventModLookup->EventModLookup.get(
+      ~sighash=topic0,
+      ~contractAddressMapping=contractInterfaceManager.contractAddressMapping,
+      ~contractAddress=log.address,
     ) {
-    | Error(exn) =>
-      logger->Logging.childErrorWithExn(exn, "Failed to parse event from RPC. Double c")
-      exn->raise
-    | Ok(res) => res
+    | None => {
+        let logger = Logging.createChildFrom(
+          ~logger,
+          ~params={
+            "chainId": chainId,
+            "blockNumber": block.number,
+            "logIndex": log.logIndex,
+            "contractAddress": log.address,
+            "topic0": topic0,
+          },
+        )
+        %raw(`null`)->ErrorHandling.mkLogAndRaise(~msg="Failed to lookup registered event", ~logger)
```

Comment:
Why do you prefer raw null over a matchable/searchable exception?

Reply:
Should I create a dummy exception for this case?

Reply:
I don't really mind but I would have probably defined a unique rescript exception and passed it in. Or perhaps add an API to errorHandling that doesn't take an exception but just throws the message. Either way the abstraction is obviously not fitting here.